common/frame_session.go

```diff
@@ -259,12 +259,29 @@ func (fs *FrameSession) initEvents() {
 					fs.onAttachedToTarget(ev)
 				case *target.EventDetachedFromTarget:
 					fs.onDetachedFromTarget(ev)
+				case *cdppage.EventJavascriptDialogOpening:
+					fs.onEventJavascriptDialogOpening(ev)
 				}
 			}
 		}
 	}()
 }
 
+func (fs *FrameSession) onEventJavascriptDialogOpening(event *cdppage.EventJavascriptDialogOpening) {
+	fs.logger.Debugf("FrameSession:onEventJavascriptDialogOpening",
+		"sid:%v tid:%v url:%v dialogType:%s",
+		fs.session.ID(), fs.targetID, event.URL, event.Type)
+
+	action := cdppage.HandleJavaScriptDialog(false)
+	if event.Type == cdppage.DialogTypeBeforeunload {
+		action = cdppage.HandleJavaScriptDialog(true)
+	}
```

Comment:
Consider adding a comment here about why we're doing this `false` -> `true` change for `beforeUnload`.

Reply:
I think this is an issue in Chromium. When we try to dismiss the dialog box, nothing happens. It's the same issue that Playwright runs into with Chromium -- https://github.com/microsoft/playwright/issues/14431

Reply:
Resolved in: f598949abe8156e2d7eabbb5f5837ad1b1415c49